src/services/block.rs

```diff
@@ -24,7 +24,11 @@ fn fill_dummy_block_ids(ids: &mut Vec<i32>) {
     }
 }
 
-pub fn start<M: Messenger>(receiver: Receiver<BlockStateOperations>, messenger: M) {
+pub fn start<M: Messenger>(
+    receiver: Receiver<BlockStateOperations>,
+    _sender: Sender<BlockStateOperations>,
```

Comment:
Since all the services are called by the same macro they all now take a sender (unused, hence the underscore). I think it's reasonable to give them all the ability to send messages to themselves. We'll see how this pattern turns out